### ISSUE TITLE
Some testcases are failed for end2end test in Brocade sim-s agent

### ISSUE THREAD

**Issue by tanshaolong** [pywbem/pywbem#1745]
Hi,

We deployed a brocade smi-s agent environment for pywbem end2end test, but some testcases are failure. The below is some details for the test.

Test Environment:
Because we can’t get the latest brocade agent software from its official site, the agent version is “12.4.0”. The detail info for the agent is here:
instance of Brocade_Agent {
   SystemCreationClassName = "Brocade_ManagementServer";
   SystemName = "bocloud-PC";
   CreationClassName = "Brocade_Agent";
   Name = "bocloud-PC_Agent";
   Version = "12.4.0 build 182";
   Date = "20150408_1949";
   ManagementServerVersion = "12.4.0 build 182";
   ManagementServerBuildDate = "20150408_1949";
   StartTime = "四月 26, 2019 11:37:22 上午 CST";
   MaxHeap = "1536m";
   LogLevel = "800";
   AuthenticationEnabled = true;
   ServerUserName = "Administrator";
   UsedMemory = "228589568";
};

Failure test cases:
The failed testcases are 18. Summarized, the failed testcases have 4 types.
1.Undefined profiles and 1 failed testcase. The error message is:
```
tests/end2endtest/test_profile_generic.py:107: in test_undefined_profiles
    undefined_profile_lines))
E   AssertionError: Server srv1 at https://192.168.2.75:5989 advertises the following profiles that are not defined in profiles.yml. This may be caused by incorrectly implemented profile names:
E   SNIA u'LaunchInContext' 1.0.0
E   SNIA u'FabricVirtualFabrics' 1.5.0
```
2.Invalid numeric value 'WBEMSolutions_ObjectManager' and 15 failed testcases. The below is just a error sample. If you need the more error info, please let me know.
```
tests/end2endtest/test_profile_snia_server.py:233: in test_System
    far_class='CIM_System')
tests/end2endtest/utils/assertions.py:763: in assert_association_a6
    _inst = conn.GetInstance(path, asserted=True)
tests/end2endtest/utils/utils.py:276: in GetInstance
    return self._call_func('GetInstance', *args, **kwargs)
tests/end2endtest/utils/utils.py:251: in _call_func
    self.raise_as_assertion_error(exc, funcname, *args, **kwargs)
tests/end2endtest/utils/utils.py:264: in raise_as_assertion_error
    exc.__class__.__name__, exc, parm_str))
E   AssertionError: Server srv1 at https://192.168.2.75:5989: WBEMConnection.GetInstance() failed and raised CIMXMLParseError - Invalid numeric value 'WBEMSolutions_ComputerSystem'
E   CIM-XML response: <?xml version="1.0" encoding="UTF-8"?><CIM DTDVERSION="2.0" CIMVERSION="2.0"><MESSAGE ID="1001" PROTOCOLVERSION="1.0"><SIMPLERSP><IMETHODRESPONSE NAME="GetInstance"><IRETURNVALUE><INSTANCE CLASSNAME="WBEMSolutions_ObjectManagerForSystem"><PROPERTY.REFERENCE PROPAGATED="false" REFERENCECLASS="WBEMSolutions_ComputerSystem" NAME="Antecedent"><VALUE.REFERENCE><INSTANCEPATH><NAMESPACEPATH><HOST>192.168.2.75</HOST><LOCALNAMESPACEPATH><NAMESPACE NAME="interop"/></LOCALNAMESPACEPATH></NAMESPACEPATH><INSTANCENAME CLASSNAME="WBEMSolutions_ComputerSystem"><KEYBINDING NAME="CreationClassName"><KEYVALUE VALUETYPE="numeric" TYPE="">WBEMSolutions_ComputerSystem</KEYVALUE></KEYBINDING><KEYBINDING NAME="Name"><KEYVALUE VALUETYPE="numeric" TYPE="">192.168.2.75</KEYVALUE></KEYBINDING></INSTANCENAME></INSTANCEPATH></VALUE.REFERENCE></PROPERTY.REFERENCE><PROPERTY.REFERENCE PROPAGATED="false" REFERENCECLASS="WBEMSolutions_ObjectManager" NAME="Dependent"><VALUE.REFERENCE><INSTANCEPATH><NAMESPACEPATH><HOST>192.168.2.75</HOST><LOCALNAMESPACEPATH><NAMESPACE NAME="interop"/></LOCALNAMESPACEPATH></NAMESPACEPATH><INSTANCENAME CLASSNAME="WBEMSolutions_ObjectManager"><KEYBINDING NAME="CreationClassName"><KEYVALUE VALUETYPE="numeric" TYPE="">WBEMSolutions_ObjectManager</KEYVALUE></KEYBINDING><KEYBINDING NAME="SystemName"><KEYVALUE VALUETYPE="numeric" TYPE="">192.168.2.75</KEYVALUE></KEYBINDING><KEYBINDING NAME="Name"><KEYVALUE VALUETYPE="numeric" TYPE="">WBEMSolutions:21d15b7b-8a00-44ea-a3af-1b2742a29a55</KEYVALUE></KEYBINDING><KEYBINDING NAME="SystemCreationClassName"><KEYVALUE VALUETYPE="numeric" TYPE="">WBEMSolutions_ComputerSystem</KEYVALUE></KEYBINDING></INSTANCENAME></INSTANCEPATH></VALUE.REFERENCE></PROPERTY.REFERENCE></INSTANCE></IRETURNVALUE></IMETHODRESPONSE></SIMPLERSP></MESSAGE></CIM>. Parameters: (CIMInstanceName(classname='WBEMSolutions_ObjectManagerForSystem', keybindings=NocaseDict({'Dependent': CIMInstanceName(classname='WBEMSolutions_ObjectManager', keybindings=NocaseDict({'CreationClassName': 'WBEMSolutions_ObjectManager', 'SystemName': '192.168.2.75', 'Name': 'WBEMSolutions:21d15b7b-8a00-44ea-a3af-1b2742a29a55', 'SystemCreationClassName': 'WBEMSolutions_ComputerSystem'}), namespace='interop', host='192.168.2.75'), 'Antecedent': CIMInstanceName(classname='WBEMSolutions_ComputerSystem', keybindings=NocaseDict({'CreationClassName': 'WBEMSolutions_ComputerSystem', 'Name': '192.168.2.75'}), namespace='interop', host='192.168.2.75')}), namespace='interop', host=None))
```
3.Instance is not found and 1 failed testcase. The error massage is:
```
tests/end2endtest/test_profile_snia_server.py:233: in test_System
    far_class='CIM_System')
tests/end2endtest/utils/assertions.py:691: in assert_association_a5
    _inst = conn.GetInstance(path, asserted=True)
tests/end2endtest/utils/utils.py:276: in GetInstance
    return self._call_func('GetInstance', *args, **kwargs)
tests/end2endtest/utils/utils.py:251: in _call_func
    self.raise_as_assertion_error(exc, funcname, *args, **kwargs)
tests/end2endtest/utils/utils.py:264: in raise_as_assertion_error
    exc.__class__.__name__, exc, parm_str))
E   AssertionError: Server srv1 at https://192.168.2.75:5989: WBEMConnection.GetInstance() failed and raised CIMError - 6 (CIM_ERR_NOT_FOUND): The instance wbemsolutions_computersystem:creationclassname=WBEMSolutions_ComputerSystem:name=192.168.2.75 does not exist.. Parameters: (CIMInstanceName(classname='WBEMSolutions_ComputerSystem', keybindings=NocaseDict({'CreationClassName': 'WBEMSolutions_ComputerSystem', 'Name': '192.168.2.75'}), namespace='interop', host='192.168.2.75'))
```
4.Profile is incorrectly swapped and 1 failed testcase. The error message is:
```
tests/end2endtest/test_profile_snia_smis.py:58: in test_snia_smis_to_profile_not_swapped
    SPEC_ORG, SPEC_NAME, len(profiles_swapped)))
E   AssertionError: Server srv1 at https://192.168.2.75:5989: The instance representing the SNIA 'SMI-S' specification references 11 profiles via CIM_ElementConformsToProfile using incorrectly swapped roles so that ConformantStandard is on the profile side
```

If you need other test detail, please let me know. Thank you

Best regards
Ray

**Comment by KSchopmeyer** [comment]
Note. Thanks for the feedback .  It is good to see users start to use that End2End test code. However, keep in mind that the goal of the end-end testing was to test pywbem code, not to develop a comprehensive definition of the SMI profiles.  All that code is in our development test environment, and not released as part of the PyPi release. We did this primarily as part of the development of the methods for get_central_instance, etc. (i.e. some of the WBEMServer class) since we test that against a wide range of servers in the SNIA testbed.

We will evaluate this to see if it is something we can fix in the profile definitions.  However, we did that work primarily to provide a platform for testing pywbem and not to develop an official definition of profiles.  Pywbem's goal is to be profile agnostic and stay away from incorporating things specific to particular profiles or some of the idiosyncrasies of particular platforms (Note that we already violated the second goal several times by adding code to account for strange exceptions, etc. that are not within the scope of the DMTF and SNIA specifications but we are trying to limit that effort).

We may use it as the basis for other work in the future but we do not want to take responsibility for
